src/deluge/playback/mode/session.cpp

```diff
@@ -560,6 +560,13 @@ void Session::doLaunch(bool isFillLaunch) {
 			}
 		}
 
+		// Arm it again if a ONCE clip, so it stops at the launchEvent
+		if (!isFillLaunch && (clip->activeIfNoSolo || clip->soloingInSessionMode)
```

Comment:
Can it be a fill launch if it's a once clip?

Reply:
This is some hack implemented by 0beron, which calls doLaunch() method outside the regular end-of-loop moment. That time when doLaunch is called with isFillLaunch=true, it is meant to only evaluate fill clips and nothing more because it was scheduled to happen in the middle of a clips loop. So not wanting to start Once clips in the start of measure 4 in a 4 measures clips loop because a 1-measure Fill clip was scheduled to launch